Add return type hint to ViewModel::getIterator()

func0der opened the pull request:
This is to satisfy static code analytic tools like PHPStan and Psalm.

Resolves #73 and also helps with #56

gsteel reviewed (changes requested):
Thanks @funcoder 👍

Really, this should be `Traversable<int, ModelInterface>` to fit with the return type of `IteratorAggregate` - when we eventually get return types added in 3.0, This will receive `Traversable` as its native return type.

Also, can you please update and target the latest minor `2.31.x`. Your change is not a BC break so it can be released in a minor.

gsteel commented:
I should also mention that to keep psalm happy here, you'll need to annotate `$this->children` as `list<ModelInterface>`

func0der commented:
:roll_eyes: Github interface confused me with the branches xD 

I hope I got everything the way you wanted it to be, @gsteel ?

func0der committed:
Add return type hint to ViewModel::getIterator()

This is to satisfy static code analytic tools like PHPStan and Psalm.

Resolves #73

Signed-off-by: func0der <529819+func0der@users.noreply.github.com>

gsteel reviewed (approved):
LGTM! Thanks @func0der
